lib/util/index.js

```diff
@@ -503,10 +501,36 @@ function hidePrivatePropertiesInClass(klass) {
     constructor(...args) {
       super(...args);
       hidePrivateProperties(this);
+      hidePrototypeProperties(this, klass);
     }
   };
 }
 
+/**
+ * Hide a property of an object.
+ * @param {object} object
+ * @param {string} name
+ */
+function hideProperty(object, name) {
+  const descriptor = Object.getOwnPropertyDescriptor(object, name);
+  descriptor.enumerable = false;
+  Object.defineProperty(object, name, descriptor);
+}
+
+/**
+ * Hide the prototype properties of an object.
+ * @param {object} object
+ * @param {Function} klass
+ */
+function hidePrototypeProperties(object, klass) {
+  const { prototype } = klass;
+  Object.getOwnPropertyNames(prototype).filter(name => {
+    return typeof prototype[name] !== 'function';
```

Comment:
This seems a little too general. What if we want to allow a public class with an ES6 getter on the prototype? Or what about classes like WebSocket that define `WebSocket.prototype.CLOSING` and other properties? I know we don't have these today, but I was thinking it would be better to apply only a fix specifically for `domain` in `lib/eventemitter.js`. WDYT?